app/controllers/inventories_controller.rb

```diff
@@ -0,0 +1,48 @@
+###
+# Copyright 2016 - 2020 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+class InventoriesController < ApplicationController
+  before_action :require_can_view_projects!
+  before_action :require_can_edit_projects!, only: [:edit, :update]
+  before_action :set_inventory, only: [:show, :update, :edit]
+  before_action :set_project, only: [:edit, :update]
+
+  include ArelHelper
+
+  def edit
+  end
+
+  def update
+    @inventory.update(inventory_params)
+    respond_with(@inventory, location: project_path(@project))
+  end
+
+  private def inventory_params
+    params.require(:inventory).permit(
+      :coc_code_override,
+    )
+  end
+
+  private def inventory_scope
+    inventory_source.joins(:project).merge(GrdaWarehouse::Hud::Project.viewable_by(current_user))
+  end
+
+  private def inventory_source
+    GrdaWarehouse::Hud::Inventory
+  end
+
+  private def set_inventory
+    @inventory = inventory_source.find(params[:id].to_i)
```

Comment:
`inventory_scope` to limit to the viewable by?

Reply:
Yep.